fcrepo-kernel-api/src/main/java/org/fcrepo/kernel/api/models/FedoraResource.java

```diff
@@ -257,4 +257,10 @@ void replaceProperties(final IdentifierConverter<Resource, FedoraResource> idTra
      * @return the resource being described
      */
     FedoraResource getDescribedResource();
+
+    /**
+     * Get a graph resource corresponding to this resource
+     * @return a graph resource
+     */
+    Resource graphResource(Converter<Resource, String> idTranslator);
```

Comment:
A lot of what I was talking about is in there, but some is not and some is not addressed in the way I meant (not your fault, my fault for using unclear language). What do you say to me sending you a PR to this PR with my highly opinionated opinions? Then you can cherrypick from that what you like.